crates/nu-utils/src/filesystem.rs

```diff
@@ -83,6 +88,35 @@ pub fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
     }
 }
 
+#[cfg(all(unix, target_os = "android"))]
+pub fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
+    // on android checking access to a directory by userid and group is not
+    // enough, it is also granted by android NDK, so to check if a user has
+    // permission just cd to that directory and check
+    let old_pwd = std::env::current_dir();
+
+    if let Err(_e) = std::env::set_current_dir(&dir) {
```

Comment:
Similarly, `set_current_dir` here will affect all the whole process and all threads. Is there a different way to check for permissions?

Reply:
I guess try to read that dir can test permission without set current directory there.

Reply:
```bash
$ chmod 0666 t
$ eza -T t
t
├── r
└── s
$ eza t
r  s
$ cd t
bash: cd: t: Permission denied
```

Reply:
`$chmod 0666 t` does not set 'x' permission?